keys_storage.go

```diff
@@ -72,6 +95,58 @@ func (s *KeysStorageInMemory) DeleteMk(pubKey Key, msgNum uint) error {
 	return nil
 }
 
+// See KeysStorage.
+func (s *KeysStorageInMemory) TruncateMks(sessionID SessionID, maxKeys int) error {
+	var seqNos []uint
+	// Collect all seq numbers
+	for _, keys := range s.keys {
+		for _, inMemoryKey := range keys {
+			if bytes.Equal(inMemoryKey.sessionID, sessionID) {
+				seqNos = append(seqNos, inMemoryKey.seqNum)
+			}
+		}
+	}
+
+	// Nothing to do if we haven't reached the limit
+	if len(seqNos) <= maxKeys {
+		return nil
+	}
+
+	// Take the sequence numbers we care about
+	sort.Slice(seqNos, func(i, j int) bool { return seqNos[i] < seqNos[j] })
+	toDeleteSlice := seqNos[0 : len(seqNos)-maxKeys]
+
+	// Put in map for easier lookup
+
+	toDelete := make(map[uint]bool)
+
+	for _, i := range toDeleteSlice {
+		toDelete[uint(i)] = true
```

Comment:
seems like leftover after latest update:
```suggestion
		toDelete[i] = true
```

also, could be optimized for much less-memory usage by replacing `bool` with empty structs (`struct{}`), if needed

Reply:
I think it's ok, this implementation is not really meant to be used as being in-memory only, I think it was just pushed out as a poc.